gridtools/shared/mosaicobj.py

```diff
@@ -0,0 +1,91 @@
+from typing import Optional, Dict
+from dataclasses import dataclass,field
+import sys
+import os
+import xarray as xr
+import numpy as np
+import numpy.typing as npt
+
+from gridtools_lib import GridObj
+
+@dataclass
+class MosaicObj:
+    mosaic_file: str = None
+    output_file: str = None
+    ntiles: int = None
+    ncontact: int = None
+    mosaic_name: str = None
+    gridlocation: str = None
+    gridfiles: npt.NDArray[np.str_] = None
+    gridtiles: npt.NDArray[np.str_] = None
+    contacts: npt.NDArray[np.str_] = None
+    contact_index: npt.NDArray[np.str_] = None
+    mosaic: object = field(init=False)
```

Comment:
sorry, one more, could we got with dataset instead of mosaic?  so that users know it's an xarray datset